File: 009/Problem9.mjs
import { performance } from "perf_hooks";

function problem() {
  var a, b;
  var c = 0;
  var s = 1000;
  var found = false;
  for (a = 1; a < s / 3; a++) {
    for (b = a; b < s / 2; b++) {
      c = s - a - b;
      if (a * a + b * b === c * c) {
        found = true;
        break;
      }
    }
    if (found) {
      break;
    }
  }
  console.log(a * b * c);
}

var start = performance.now();
for (var i = 0; i < 1000; i++) {
  problem();
}
var end = performance.now();
console.log("Function took " + (end - start) / 1000);

File: 009/Problem9.rs
fn main() {
    let start = std::time::SystemTime::now();
    for _ in 0..1000 {
        problem();
    }
    match start.elapsed() {
        Ok(elapsed) => {
            println!("Function took: {}", elapsed.as_nanos())
        }
        Err(e) => {
            println!("Error: {:?}", e)
        }
    }
}

fn problem() {
    let mut c: usize;
    let s: usize = 1000;
    let mut ans: usize = 10;
    for a in 1..s/3 {
        for b in a..s/2 {
            c = s - a - b;

            if a * a + b * b == c * c {
                ans = a * b * c;
                break;
            }
        }
    }
    println!("{}", ans);
}
